system/CodeIgniter.php

```diff
@@ -345,13 +355,22 @@ public function run(?RouteCollectionInterface $routes = null, bool $returnRespon
             // If the route is a 'redirect' route, it throws
             // the exception with the $to as the message
             $this->response->redirect(base_url($e->getMessage()), 'auto', $e->getCode());
+
+            if ($this->returnRespose) {
+                return $this->response;
+            }
+
             $this->sendResponse();
 
             $this->callExit(EXIT_SUCCESS);
```

Comment:
Not related to this PR, but is this `callExit()` necessary?

Reply:
I'm not sure - maybe just to enforce a "never return" condition?

Reply:
If you will add `@phpstan-return never` to `protected function callExit`, then phpstan correctly flags that the `return;` below will be reached. So, this must be a bug?